pyuvdata/tests/test_uvdata.py

```diff
@@ -661,3 +661,172 @@ def test_reorder_pols():
     uvtest.checkWarnings(uv1.read_uvfits, [testfile], message='Telescope EVLA is not')
     uv2.reorder_pols()
     nt.assert_equal(uv1, uv2)
+
+
+def test_add():
+    uv_full = UVData()
+    testfile = os.path.join(DATA_PATH, 'day2_TDEM0003_10s_norx_1src_1spw.uvfits')
+    uvtest.checkWarnings(uv_full.read_uvfits, [testfile],
+                         message='Telescope EVLA is not')
+
+    # Add frequencies
+    uv1 = copy.deepcopy(uv_full)
+    uv2 = copy.deepcopy(uv_full)
+    uv1.select(freq_chans=np.arange(0, 32))
+    uv2.select(freq_chans=np.arange(32, 64))
+    uv1 += uv2
+    # Check history is correct, before replacing and doing a full object check
+    nt.assert_equal(uv_full.history + '  Downselected to specific frequencies'
+                    ' using pyuvdata. Combined data along frequency axis using'
+                    ' pyuvdata.', uv1.history)
+    uv1.history = uv_full.history
+    nt.assert_equal(uv1, uv_full)
+
+    # Add polarizations
+    uv1 = copy.deepcopy(uv_full)
+    uv2 = copy.deepcopy(uv_full)
+    uv1.select(polarizations=uv1.polarization_array[0:2])
+    uv2.select(polarizations=uv2.polarization_array[2:4])
+    uv1 += uv2
+    nt.assert_equal(uv_full.history + '  Downselected to specific polarizations'
+                    ' using pyuvdata. Combined data along polarization axis using'
+                    ' pyuvdata.', uv1.history)
+    uv1.history = uv_full.history
+    nt.assert_equal(uv1, uv_full)
+
+    # Add times
+    uv1 = copy.deepcopy(uv_full)
+    uv2 = copy.deepcopy(uv_full)
+    times = np.unique(uv_full.time_array)
+    uv1.select(times=times[0:len(times) / 2])
+    uv2.select(times=times[len(times) / 2:])
+    uv1 += uv2
+    nt.assert_equal(uv_full.history + '  Downselected to specific times'
+                    ' using pyuvdata. Combined data along baseline-time axis using'
+                    ' pyuvdata.', uv1.history)
+    uv1.history = uv_full.history
+    nt.assert_equal(uv1, uv_full)
+
```

Comment:
I treat the baseline-time axis as a single entity to add. In the testing, it was easiest to trim/add by times. I'll add another one that does it by baseline for completeness.